src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1827,22 +1845,23 @@ export default class AircraftModel {
      * @return {number}
      */
     _calculateTargetedAltitudeVnavDescent(hardRestrictedWaypointModel) {
-        let waypointAltitude = hardRestrictedWaypointModel.altitudeMaximum;
-        const softOrHardRestrictedWaypoint = this.fms.nextAltitudeRestrictedWaypoint;
-        const nextFixIsSoftlyRestricted = softOrHardRestrictedWaypoint.name !== hardRestrictedWaypointModel.name;
-        const nextFixIsAtOrAboveRestriction = softOrHardRestrictedWaypoint.altitudeMinimum !== INVALID_NUMBER;
-
-        // NOTE: Currently does not cover any "AT/BELOW", since we will already be descending to bottom anyway
-        if (nextFixIsSoftlyRestricted && nextFixIsAtOrAboveRestriction) {
-            waypointAltitude = softOrHardRestrictedWaypoint.altitudeMinimum;
-        }
-
         // TODO: This could be improved by making the descent at the exact rate needed to reach
         // the altitude at the same time as reaching the fix. At this point, the problem is that
         // while we DO know the descent rate and descent angle to shoot for, we don't know the
         // length of time before the next update, so we can't accurately estimate the altitude to
         // target in the current iteration.
-        return Math.max(waypointAltitude, this.mcp.altitude);
+
+        const nextRestrictedWaypoint = this.fms.nextAltitudeRestrictedWaypoint;
+        if (_isNil(nextRestrictedWaypoint)) {
+            return this.fms.getBottomAltitude();
+        }
+
+        if (nextRestrictedWaypoint.altitudeMinimum !== INVALID_NUMBER) {
+            return nextRestrictedWaypoint.altitudeMinimum;
+        }
+        else {
```

Comment:
`else` is not needed here.  

Also, `else` should be on the same line as the closing bracket of the previous logical statement:

```javascript
// undesired
if (condition)
{
  // do stuff
}
else
{
  // do other stuff
}

if (condition) {
  // do stuff
}
else {
  // do other stuff
}

// desired
if (condition) {
  // do stuff
} else {
  // do other stuff
}

if (condition) {
  // do stuff
  return
}
```

Be sure you are using `eslint` with your editor: https://eslint.org/docs/user-guide/integrations.  It will warn you about these style inconsistencies.